### PR TITLE
#166790004 Users Should Follow Each Other

### PR DESCRIPTION
**Pull request by IsaiahRn**
### What does this PR do?

- Users should be able to follow each other

#### Description of Task to be completed

- create follow endpoints

- verify the endpoint to allow only valid users

- make sure users cannot follow themselves

- add tests for the endpoints

- document feature using swagger

#### How should this be manually tested?

1. Run the following commands in the terminal.
```
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-Users-follow-unfollow-166790004
npm install
npm start
```

2. Test endpoints, via Postman:

- Follow and unfollow a user `http://localhost:3000/api/:username/follow`

- Fetch the user's followers `http://localhost:3000/api/followers`

- Fetch the user's following `http://localhost:3000/api/following`

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?
[#166790004](https://www.pivotaltracker.com/story/show/166790004)
#### Screenshots (if appropriate)

- Follow a user:
Screen Shot 2019-07-19 at 11 54 10

- Unfollow a user
Screen Shot 2019-07-19 at 11 54 29

- Fetch followers
Screen Shot 2019-07-24 at 12 42 29

- Fetch following
Screen Shot 2019-07-24 at 12 42 15

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by niyoceles — DISMISSED** [review]

File: controllers/followers.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      attributes: ['createdAt', 'updatedAt'],
+      where: { followed: loggedinUser.id },
+      include: [{ model: users, as: 'User', attributes: ['username', 'image'] }]
+    });
+    return followers.length
+      ? res.status(200).json({
+        message: 'Followers',
+        followers
+      })
+      : res.status(404).json({ errors: "You don't have followers" });
```
Comment (kodek-sleuth) [inline controllers/followers.js]: Please use `error` not `errors`

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

File: tests/followers.test.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+let tokenGen;
+let userName1;
+let userName2;
+const userNameSample = 'isaie';
+
+describe('Following a user', () => {
```
Comment (kodek-sleuth) [inline tests/followers.test.js:15]: Please use seeders to further simplify your tests

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by kodek-sleuth — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by niyoceles — DISMISSED** [review]

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by kodek-sleuth — APPROVED** [review]

**Review by niyoceles — APPROVED** [review]

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Pull request merged**